src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -639,3 +641,52 @@ ErrorIfAnyMetadataNodeOutOfSync(List *metadataNodeList)
 		}
 	}
 }
+
+
+/*
+ * IsWorkerTheCurrentNode checks if the given worker refers to the
+ * the current node by comparing the server id of the worker and of the
+ * current nodefrom pg_dist_node_metadata
+ */
+bool
+IsWorkerTheCurrentNode(WorkerNode *workerNode)
+{
+	int connectionFlags = REQUIRE_METADATA_CONNECTION;
+
+	MultiConnection *workerConnection =
+		GetNodeUserDatabaseConnection(connectionFlags,
+									  workerNode->workerName,
+									  workerNode->workerPort,
+									  CurrentUserName(),
+									  NULL);
+	const char *command =
+		"SELECT metadata ->> 'server_id' AS server_id FROM pg_dist_node_metadata";
```

Comment:
The logic to parse the output of a json is too complex. To remove those codes, can we do
```SQL
SELECT count(*) FROM pg_dist_node_metadata where metadata = %s::jsonb; 
```
where we supply our own pg_dist_node_metadata, which can be get:
```C
		metadataJsonbDatum = DistNodeMetadata();
		metadataJsonbStr = DatumGetCString(DirectFunctionCall1(jsonb_out,
															   metadataJsonbDatum));
```

If so, I'd rather prefer to parse (or even check a single row returns or not)

Reply:
(Oh, we mostly copy & pasted from other places, let me have a look again first)

Reply:
yes, majority of the code already exists, sorry, I couldn't realize  :)